### PR TITLE
PLAT-375: test no dead lock on semaphore

### PR DESCRIPTION
**Pull request by ivanshibitov**
1. Add test on simple dead lock detection on parent semaphore
2. tests skipped since I must be changed after PLAT-432: parent semaphore must be set to 1

### CODE REVIEW

File: ledger-core/virtual/integration/contract_calls_test.go [hunk 1]
```diff
@@ -77,6 +77,159 @@ func assertVCallRequest(t *testing.T,
...
+				})
+
+				typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+					switch res.Callee {
+					case objectAGlobal:
+						require.Equal(t, []byte("finish A.Foo"), res.ReturnArguments)
+						assertVCallResult(t, res, server.GlobalCaller(), objectAGlobal, test.flagsA, outgoingA)
+					case objectBGlobal:
+						require.Equal(t, []byte("finish B.Bar"), res.ReturnArguments)
+						assertVCallResult(t, res, objectAGlobal, objectBGlobal, test.flagsA, outgoingA)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/contract_calls_test.go:202]: flagsB ?

File: ledger-core/virtual/integration/contract_calls_test.go [hunk 2]
```diff
@@ -77,6 +77,159 @@ func assertVCallRequest(t *testing.T,
 
 }
 
+func Test_NoDeadLock_WhenOutgoingComeToSameNode(t *testing.T) {
+	t.Skip("https://insolar.atlassian.net/browse/PLAT-432")
```
Comment (ruz) [inline ledger-core/virtual/integration/contract_calls_test.go:81]: no t.Log with link to test case in TR.

Comment (ruz) [reply ledger-core/virtual/integration/contract_calls_test.go:81]: I see cases logged below, this skip should be placed after logging test case.

Comment (ivanshibitov) [reply ledger-core/virtual/integration/contract_calls_test.go]: done

**Pull request merged**